Fixes shadow atlas for empty scene

Kuranes opened the pull request:
Handles empty scene shadowing using shadow atlas

cedricpinson commented:
It's not clear why we need to do something when we dont draw, could you have a look plz, maybe it's not needed ?

Kuranes commented:
- documented how/what/why empty scene are handled that way
- added case of empty scene
- renamed few method for clarity
- removed useless uniform change in the no scene case

cedricpinson commented on sources/osgShadow/ShadowMap.js:
```diff
@@ -43,14 +43,18 @@ utils.createPrototypeObject(
             var near = nv.getComputedNear(),
                 far = nv.getComputedFar();
 
+            if (near === Infinity && far === -Infinity) {
```
are you sure it's && and not a ||

Kuranes replied:
yes.
It checks for exact state  'no geom at all did pass the cullvisitor cull'.
So it check the exact state  CullVisitor resets near and far to before doing a cull. if either is not '===' it would mean something has passed the test and a geom is drawn.
